### PR TITLE
#187419006 Implement Reset Password

### PR DESCRIPTION
**Pull request by Hakizimana-Clement**
#### What does this PR do?
This PR adds a feature that allows users to reset their forgotten passwords.

#### Description of Task to be completed?
- User click on "Forgot password" and get email which contain token for reseting password.
- After to click on it, its redirect to page where user enter his/her password along with confirm password in form with validation.

#### How should this be manually tested?
1. Navigate to the login page.
2. Click on the 'Forgot Password?' link.
3. Enter the associated email address and submit.
4. Check the email for the reset password token.
5. Follow the link to the reset password page.
6. Enter and submit the new password.

#### What are the relevant pivotal tracker/Trello stories?
#187419006

### CODE REVIEW

**Review by cyusasnave — APPROVED** [review]

**Review by Tuyisenge2 — APPROVED** [review]

File: src/components/ResetPassword/ResetPasswordForm.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		if (!token) {
+			showErrorMessage('Invalid token. please try again');
+			return navigate('/login');
+		}
+		try {
+			const { password } = data;
+			const res = await dispatch(resetPassword({ password, token })).unwrap();
+			showSuccessMessage(res.message);
+			setTimeout(() => {
+				navigate('/login');
```
Comment (leandreAlly) [inline src/components/ResetPassword/ResetPasswordForm.tsx]: remove this timeout function

File: src/components/ResetPassword/ResetPasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			showErrorMessage(
+				err?.data?.message ||
+					err?.message ||
+					'Unknown error occured! Please try again!',
+			);
+		}
+	};
+
+	const navigate = useNavigate();
+	const handleNavigation = () => {
```
Comment (leandreAlly) [inline src/components/ResetPassword/ResetPasswordForm.tsx]: put this useNavigate() hook at the top of the page

File: src/components/ResetPassword/ResetPasswordForm.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+				<FormInput
+					type="password"
+					placeholder="Confirm Password"
+					otherStyles="text-center bg-neutral-white border border-overlay w-full rounded-sm py-1 mobile:py-2 mb-2"
+					errorStyles="text-xs tablet:text-[0.73rem]"
+					{...register('confirmPassword')}
+					error={errors.confirmPassword}
+				/>
+				<button
+					disabled={isSubmitting}
```
Comment (leandreAlly) [inline src/components/ResetPassword/ResetPasswordForm.tsx]: use our reusable button components.

File: src/components/Layouts/ForgottonPasswordContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import backgroundImage from '../../assets/register-login-background.svg';
+import logo from '../../assets/otherLogo.svg';
+import { ReactNode } from 'react';
```
Comment (leandreAlly) [inline src/components/Layouts/ForgottonPasswordContainer.tsx:1]: we no longer handing asset in this way, check the file called image and import logo from there. and also always organize your import.

File: src/components/Layouts/ResetPasswordContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import backgroundImage from '../../assets/register-login-background.svg';
+import logo from '../../assets/otherLogo.svg';
+import { ReactNode } from 'react';
```
Comment (leandreAlly) [inline src/components/Layouts/ResetPasswordContainer.tsx:1]: same case here, import your asset from main file, image.ts

**Review by leandreAlly — CHANGES_REQUESTED** [review]

File: src/components/ResetPassword/ResetPasswordForm.tsx (hunk 1 above)
Comment (Hakizimana-Clement) [reply src/components/ResetPassword/ResetPasswordForm.tsx]: let me remove it.

File: src/components/Layouts/ForgottonPasswordContainer.tsx (hunk 4 above)
Comment (Hakizimana-Clement) [reply src/components/Layouts/ForgottonPasswordContainer.tsx:1]: let me change it

File: src/components/Layouts/ResetPasswordContainer.tsx (hunk 5 above)
Comment (Hakizimana-Clement) [reply src/components/Layouts/ResetPasswordContainer.tsx:1]: thanks, let me change it

File: src/components/ResetPassword/ResetPasswordForm.tsx (hunk 2 above)
Comment (Hakizimana-Clement) [reply src/components/ResetPassword/ResetPasswordForm.tsx]: let me put on the top of the page

File: src/components/ResetPassword/ResetPasswordForm.tsx (hunk 3 above)
Comment (Hakizimana-Clement) [reply src/components/ResetPassword/ResetPasswordForm.tsx]: let me use it

**Pull request merged**